### PR TITLE
F2127 ratefilter byboth 2

### PR DESCRIPTION
**Pull request by ruslanu**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [ + ] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [ + ] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ + ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- rate_filter track by_both now available using IPPair
-
-

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: src/detect-engine-threshold.c [hunk 1]
```diff
@@ -281,6 +319,99 @@ static inline void RateFilterSetAction(Packet *p, PacketAlert *pa, uint8_t new_a
     }
 }
 
+static int IsThresholdQualify(DetectThresholdEntry* lookup_tsh, const DetectThresholdData *td, uint32_t packet_time)
```
Comment (victorjulien) [inline src/detect-engine-threshold.c]: this function name doesn't really help me understand what it does. Can you rename and/or add a comment explaining the purpose?

File: src/detect-engine-threshold.c [hunk 2]
```diff
@@ -281,6 +319,99 @@ static inline void RateFilterSetAction(Packet *p, PacketAlert *pa, uint8_t new_a
...
+        }
+    } /* else - if (lookup_tsh->tv_timeout != 0) */
+
+    return ret;
+}
+
+static void AddEntryToHostStorage(Host *h, DetectThresholdEntry *e, uint32_t packet_time)
+{
+    if (h && e)
+    {
```
Comment (victorjulien) [inline src/detect-engine-threshold.c]: minor: style: no newline for opening brackets, except for the opening of a function

File: src/util-threshold-config.c [hunk 3]
```diff
@@ -916,6 +917,9 @@ static int ParseThresholdRule(DetectEngineCtx *de_ctx, char *rawstr,
                 parsed_track = TRACK_DST;
             else if (strcasecmp(th_track,"by_src") == 0)
                 parsed_track = TRACK_SRC;
+            else if (strcasecmp(th_track, "by_both") == 0) {
+                parsed_track = TRACK_BOTH;
```
Comment (victorjulien) [inline src/util-threshold-config.c]: if I read this correctly, the parser accepts 'by_both' for threshold, event_filter and rate_filter. However it looks like the implementation only implments rate_filter (in ThresholdHandlePacketIPPair).

So we'll either need to be stricter in the parser and reject unsupported stuff, or we need to add support for threshold/event_filter.

**Comment by victorjulien** [comment]
Can you make sure the commit messages use formatting as described here: https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Code_Submission_Quality_Criteria#Commits

Especially: some of the subject lines are too long. Also, no need to have the 'Feature XXXX' on the subject line. Instead add the subsystem you're modifying (e.g. ipair: blah or thresholding: blah) and the ticket id to the commit message body.

Thanks!

**Comment by victorjulien** [comment]
Oh a final thing, can you make sure the formatting of the PR itself is correct for the next PR? There should be 'checkboxes'.

File: src/util-threshold-config.c (hunk 3 above)
Comment (ruslanu) [reply src/util-threshold-config.c]: The parser will not accept "by_both" for threshold and event_filter due to DETECT_THRESHOLD_REGEX

**Pull request closed without merging**